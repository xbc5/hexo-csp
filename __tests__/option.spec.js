const getSandbox = require("./support/sandbox");
const { process, mockConfig, contentFor } = require("hexo-test-utils");
const sandbox = getSandbox();

async function runTest(options, metaTag) {
  const ctx = await sandbox({ fixtureName: "default" });
  mockConfig(ctx, "csp", options);
  await process(ctx);
  const content = await contentFor(ctx, "fake.html");
  expect(content.toString().trim()).toMatchSnapshot();
}

describe("when enabled:false", () => {
  it("should render the content without modification", async () => {
    await runTest({ enabled: false });
  });
});

describe("when enabled:true", () => {
  it("should render the content with modifications", async () => {
    const meta = '<meta http-equiv="" content="">';
    await runTest({ enabled: true }, meta);
  });
});
