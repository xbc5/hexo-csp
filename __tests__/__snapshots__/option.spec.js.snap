// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when enabled:false should render the content without modification 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
  <head>
    <title>Fake Title</title>
  </head>
  <body></body>
</html>"
`;

exports[`when enabled:true should render the content with modifications 1`] = `
"<!DOCTYPE html><html lang=\\"en\\"><head><meta http-equiv=\\"Content-Security-Policy\\" content=\\"\\">
    <title>Fake Title</title>
  </head>
  <body>

</body></html>"
`;
